src/containers/EncounterDetails/hooks.ts

```diff
@@ -21,11 +32,51 @@ export function useEncounterDetails(encounterId: string) {
             const encounter = sourceMap.Encounter[0];
             const patient = sourceMap.Patient[0];
             const practitioner = sourceMap.Practitioner[0];
-            const practitionerRole = sourceMap.PractitionerRole[0];
-            return { encounter, practitioner, practitionerRole, patient };
+
+            return {
+                id: encounter?.id,
+                patient,
+                practitioner: practitioner,
+                status: encounter?.status,
+                period: encounter?.period,
+                humanReadableDate: encounter?.period?.start && formatHumanDateTime(encounter?.period?.start),
+                encounter: encounter,
+            };
         });
     });
-    return encounterInfoRD;
+
+    const completeEncounter = useCallback(async () => {
```

Comment:
This callback should not depends on encounterInfoRD, pass the Encounter instance as an argument:
```javascript
const completeEncounter = useCallback(async (encounter: Encounter) => {
...
}
```

In this case you don't need `if (isSuccess(encounterInfoRD))`

And `manager.set` can be rewrite like this:
```javascript
manager.set((currentData) => ({
    ...currentData,
    encounter: saveResponse.data,
});
```